### PR TITLE
raise exception if transaction send fail

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
如果交易广播失败则需要抛失败异常

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
…

## Pull request type
 
_Put an `x` in the boxes that apply_
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

## Any other comments?
…

### CODE REVIEW

File: electrum/pywalib.py [hunk 1]
```diff
@@ -534,7 +534,11 @@ def serialize_and_send_tx(self, tx_dict, vrs):
 
     @classmethod
     def _send_tx(cls, tx_hex: str) -> str:
-        return cls.get_provider().broadcast_transaction(tx_hex).txid
+        receipt = cls.get_provider().broadcast_transaction(tx_hex)
+        if not receipt or not receipt.is_success:
+            raise Exception(f"Transaction send fail. error_message: {receipt.receipt_message}", tx_hex)
```
Comment (taimanhui) [inline electrum/pywalib.py:539]: `not receipt`的话哪来`receipt.receipt_message`？

**Review by taimanhui — COMMENTED** [review]
有处小问题

File: electrum/pywalib.py (hunk 1 above)
Comment (huazhouwang) [reply electrum/pywalib.py:539]: ack, `not receipt` 判断没有必要，我去掉吧

File: electrum/pywalib.py [hunk 2]
```diff
@@ -534,7 +534,11 @@ def serialize_and_send_tx(self, tx_dict, vrs):
 
     @classmethod
     def _send_tx(cls, tx_hex: str) -> str:
-        return cls.get_provider().broadcast_transaction(tx_hex).txid
+        receipt = cls.get_provider().broadcast_transaction(tx_hex)
+        if not receipt.is_success:
```
Comment (taimanhui) [inline electrum/pywalib.py:538]: 看到就说下，一般两个逻辑并列的话选择那个不用否定的版本。不过也无所谓纠结这些，这里说想突出错误处理也行那就不算并列了

**Review by taimanhui — APPROVED** [review]

File: electrum/pywalib.py (hunk 2 above)
Comment (huazhouwang) [reply electrum/pywalib.py:538]: 嗯，我一般会把抛出异常放前面，正常返回放后面

**Review by shuaiLiWang — APPROVED** [review]

**Pull request merged**